app/src/androidTest/java/org/mozilla/fenix/helpers/MockLocationUpdatesRule.kt

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.helpers
+
+import android.content.Context
+import android.location.Location
+import android.location.LocationManager
+import android.os.Build
+import android.os.SystemClock
+import android.util.Log
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import org.junit.rules.ExternalResource
+import java.util.Date
+import kotlin.random.Random
+import org.mozilla.fenix.helpers.TestHelper.executeShellCommandBlocking
+
+private const val mockProviderName = LocationManager.GPS_PROVIDER
+
+/**
+ * Rule that sets up a mock location provider that can inject location samples
+ * straight to the device that the test is running on.
+ *
+ * Credit to the mapbox team
+ * https://github.com/mapbox/mapbox-navigation-android/blob/87fab7ea1152b29533ee121eaf6c05bc202adf02/libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/MockLocationUpdatesRule.kt
+ *
+ */
+class MockLocationUpdatesRule : ExternalResource() {
+    private val instrumentation = getInstrumentation()
+    private val appContext = (ApplicationProvider.getApplicationContext() as Context)
+    val latitude = Random.nextDouble(-90.0, 90.0)
+    val longitude = Random.nextDouble(-180.0, 180.0)
+
+    private val locationManager: LocationManager by lazy {
+        (appContext.getSystemService(Context.LOCATION_SERVICE) as LocationManager)
+    }
+
+    override fun before() {
+        /* ADB command to enable the mock location setting on the device.
+         * Will not be turned back off due to limitations on knowing its initial state.
+         */
+        instrumentation.uiAutomation.executeShellCommandBlocking(
```

Comment:
Aside I wonder if we should use this function in https://github.com/mozilla-mobile/fenix/blob/main/app/src/androidTest/java/org/mozilla/fenix/helpers/HomeActivityTestRule.kt#L81